### ISSUE TITLE
new CLI doc option: reorder docs by checking front matter of all docs in the repository regardless of whether there was/wasn't a change

### ISSUE THREAD

**Issue by janiksam** [readmeio/rdme#731]
Hello team,

Our flow is to update the OpenAPI spec through GitHub actions and then update the order of the endpoints using front matter found in corresponding markdown files in the repository. 

I would like to request a `force` option in the `docs` method for the engine to go through all the docs in the master branch and update the order of my endpoints by looking at the front matter of the markdown files. In other words, I would like the CLI to look at all the docs as opposed to only docs that have changed or were updated since the last sync to readme.

Something along the lines of (the option could be named differently of course other than `--force`, for example, `--re-order`, `--all`)

```
- name: Sync docs to ReadMe
        if: github.event_name == 'push' && github.event.ref == 'refs/heads/master'
        uses: readmeio/rdme@8.4.0
        with:
          rdme: docs ./docs --key=${{ secrets.README_API_KEY }} --version=${{ secrets.README_MAIN_VERSION }} --force
```

Please let me know if you need any additional information. We would greatly appreciate your help.

Thanks!
Sam

**Comment by kanadgupta** [comment]
Hey @janiksam! Thanks for writing in.

I'm having trouble understanding what this flag would be doing that isn't already being done in the current behavior of the `docs` command[^1]. The command already scans through every Markdown file within the directory you specify. If the slug of a given Markdown file matches an existing page in ReadMe, it will update the page as opposed to creating a new one, but every page with Markdown + front matter is already being looked at. Is this your understanding of the `docs` command?

Please let me know if I'm misunderstanding your request!

[^1]: Everything described here also applies to the `changelogs` and `custompages` commands as well.

**Comment by janiksam** [comment]
Hey @kanadgupta 

Thank you for getting back quickly and answering my question. That is exactly the clarification I was looking for.

However, I have a case where the Github actions ran successfully, but when I go back to readme to the version of the documentation that is being updated by the sync file, the files are ordered in some other sequence. 

Please note the markdown files have front matter with the `order` parameter indicating which order these documents should be rendered. The documents can be found [here](https://github.com/voucherifyio/voucherify-openapi/tree/master/docs/reference-docs).

I will provide an example for more clarification. In this example, the intended sequence is the following:

**Validations API**
- [ ] Establish Validation Session (front matter `order`: 1 )
- [ ] Validate Voucher (front matter `order`: 2)
- [ ] Validate Voucher (client-side) (front matter `order`: 3)
- [ ] Validate Promotions (front matter `order`: 4)
- [ ] Validate Promotion Tier (front matter `order`: 5)

When I render the documentation, I get the following sequence:

<img width="242" alt="Screenshot 2023-01-12 at 10 03 22" src="https://user-images.githubusercontent.com/95409292/212025823-ea31df17-baf7-47f0-b0b4-add7c935e05d.png">

The document titled Establish Validation Session was placed second in the sequence versus the first. 

Please note this behaviour is true in the other sections of the API Reference documentation, independent of the document type (link, basic, endpoint). 

Please let me know if you need any additional information. We would greatly appreciate your help.

Thanks!
Sam

**Comment by kanadgupta** [comment]
Hi @janiksam! Are you also syncing these "Validations API" endpoint pages via an API definition (OpenAPI, Swagger, etc.)? If so, could you provide a link to that file? It's possible that the page order/hierarchy is getting set in one way via your Markdown files and another way via your API definition. This docs page provides a good explanation of how we determine the page order/hierarchy based on the structure of your API definition: https://docs.readme.com/main/docs/openapi-categories-pages-subpages

I should also note that there was a long-standing known issue where re-syncing an API definition would cause certain page orders to be mixed up but we fixed this in our Jan. 10th release. Can you confirm that you're still seeing this issue?

**Comment by janiksam** [comment]
Hi @kanadgupta,

Hope you are doing well and thank you for your response.

Our API definition file can be found [here](https://raw.githubusercontent.com/voucherifyio/voucherify-openapi/master/reference/OpenAPI.json). 

Just as you have indicated, I am using the API definition file to set the order of the endpoints as close as possible to the desired order in which we would like them to be rendered. However, there are cases where we would like to reorder the endpoints. 

I'll give an example. We have endpoints that use the **GET** and **POST** method for listing resources and creating resources, respectively. Then we have a separate endpoint to retrieve a _specific_ resource only. In the API definition file, the endpoint is placed below the **POST** endpoint and there's no way to reorder it in the API definition because it is a different endpoint all together (i.e. not the same as the other two). We would like to place this **GET**-a-single-resource right below or next to the **GET**-all-resources endpoint as opposed to below the **POST**-create-a-resource endpoint.

<img width="259" alt="Screenshot 2023-01-19 at 09 09 04" src="https://user-images.githubusercontent.com/95409292/213389592-4460886f-9e4c-401b-b43d-90332fbe8130.png">

To simplify the process and eliminate the extra step to go into the Readme API (as indicated [here](https://docs.readme.com/main/docs/reordering-pages#how-ordering-works-in-our-api)) and make calls for each document to update the sequence, we have decided to use front matter. This is more convenient because we can go into files, enter new values for the `order` parameter for a set of documents under a given parent page, sync the docs using rdme and let the automation take care of the rest. 

I would like to indicate that the syncing process is continuing to give errors. Specifically, I would like to indicate we are two errors we continue to receive. 

Screenshots below:

<img width="895" alt="Screenshot 2023-01-19 at 09 22 54" src="https://user-images.githubusercontent.com/95409292/213391997-49d86e19-b739-4f2d-8a0d-68cd208a08bd.png">

<img width="862" alt="Screenshot 2023-01-19 at 09 06 04" src="https://user-images.githubusercontent.com/95409292/213392012-43804172-3247-45fe-b882-8251b21ffc9c.png">

I invite you to provide insight into the syncing file found [here](https://raw.githubusercontent.com/voucherifyio/voucherify-openapi/master/.github/workflows/sync.yml). Please note we 
1) first sync the API definition so that the order is set according to the API definition, then 
2) we sync the markdown docs with front matter to complete the re-ordering process.

We would appreciate your help with eliminating the syncing errors so that we can test the fixes for the `order` parameter.

Best,
Sam

**Comment by janiksam** [comment]
Hi @kanadgupta,

A follow up to the errors indicated in the above comment. 

**We couldn't find your API key**
In an effort to eliminate a wrong API key, I went ahead and updated the README API key from readme within GitHub secrets. Once I did that, I received the following error telling me to split up my file.

**Request timed out**
I have attempted to use the rdme CLI to split up the file using the reduce method, but unfortunately the default structure in readme creates parent pages for each set of reduced API files. This results in Categories and parent pages, when in our case the categories would be sufficient. I prefer using the URL syncing approach (as can be seen in our [sync](https://raw.githubusercontent.com/voucherifyio/voucherify-openapi/master/.github/workflows/sync.yml) file) because it worked very well in terms of syncing directly from the readme dashboard. This very nicely creates one API category and collapsable parent pages for the endpoints.

Best,
Sam
